### ISSUE TITLE
Support Ease of Access

### ISSUE THREAD

**Issue by michaelDCurran** [nvaccess/test_nvdaTracImport#23]
**Reported by jteh on 24 Apr 2009 01:58 UTC**
**Blocked by #5037**
**Blocking #100, #3854, #4158**
NVDA can currently only run on the user input desktop on which it was started. It cannot switch to secure input desktops, which is required for UAC screens and the Windows logon screen. For Windows Vista and later, the Ease of Access framework is used to launch ATs on other input desktops. NVDA should support Ease of Access.

**Comment by michaelDCurran** [comment]
**Modified by jteh on 24 Apr 2009 01:59 UTC**

**Comment by michaelDCurran** [comment]
**Comment by jteh on 18 May 2009 01:07 UTC**
The [service for Windows XP](ticket:97) works very well on Windows Vista and Windows 7. In addition, there are some issues blocking the implementation of useable support for EoA. We'll just use the service on Vista and Windows 7 unless there is a good reason to support EoA. Closing as wontfix for now; reopen later if the situation changes.

**Comment by michaelDCurran** [comment]
**Modified by jteh on 18 May 2009 01:07 UTC**

**Comment by michaelDCurran** [comment]
**Comment by jteh on 19 Jan 2014 15:05 UTC**
There are now a few good reasons to register with Ease of Access:
* Windows 8 and 8.1 intentionally delay all desktop apps set to start automatically by several seconds. This can be reduced somewhat with a registry tweak, but not entirely, and we shouldn't make changes that affect things other than NVDA anyway.
* ATs need to be registered with EoA in order to be candidates for activation with Windows+volumeUp.
* Aside from these, it's probably more future proof to do this and it means we don't have a background service running (no matter how light) even when NVDA isn't.

Windows 8 introduced changes to Ease of Access that make it quite usable for us. Specifically, you can now specify that an AT shouldn't be terminated on a desktop switch, which is exactly what we need.

The question is whether to do this for Windows Vista and 7 as well. Termination on desktop switch is annoying, but I've managed to come up with a convoluted way to work around that, though it does require another executable to be running while NVDA is running.

**Comment by michaelDCurran** [comment]
**Modified by jteh on 19 Jan 2014 15:05 UTC**
